### ISSUE TITLE
llvm: assertion triggered when casting non-register-sized integer to float

### ISSUE THREAD

**Issue by jacobly0** [ziglang/zig#17381]
```zig
export fn entry() void {
    var a: u15 = 42;
    var b: f32 = @floatFromInt(a);
    _ = b;
}
```
```
$ zig build-obj repro.zig -mcpu x86_64+avx512f
LLVM Emit Object... zig: llvm/include/llvm/CodeGen/ValueTypes.h:300: MVT llvm::EVT::getSimpleVT() const: Assertion `isSimple() && "Expected a SimpleValueType!"' failed.
Aborted (core dumped)
```

This bug is currently triggered by [this code](https://github.com/ziglang/zig/blob/412d863ba5801c1376af7ab8f04a71b839a820a6/src/resinator/cli.zig#L574) when compiling the compiler under the proper circumstances (don't look at the cast direction too closely :roll_eyes:).

upstream: llvm/llvm-project#68068

**Comment by andrewrk** [comment]
Is this a regression? I notice the fix has not made its way into the release/17.x branch.

**Comment by jacobly0** [comment]
It has been a bug since at least 16.0.0.

**Comment by andrewrk** [comment]
Gotcha, thanks for checking. Pushing a workaround shortly.

**Comment by andrewrk** [comment]
Related: #17514

Master branch behavior tests are already tripping another assertion.

**Comment by andrewrk** [comment]
Workaround landed in ab4d6bf468bd8cba4ffd2d700d83e9707f5307b1; leaving open to remove the workaround in LLVM 18.

**Comment by auxiliaire** [comment]
Hi, this is still an issue on master branch with LLVM 18 (from apt.llvm.org, Ubuntu 22.04).
```
/home/v/Projects/zig/zig/src/zig_llvm.cpp:1350:72: error: ‘WebKit_JS’ is not a member of ‘llvm::CallingConv’
 1350 | static_assert((CallingConv::ID)ZigLLVM_WebKit_JS == llvm::CallingConv::WebKit_JS, "");
      |                                                                        ^~~~~~~~~
make[2]: *** [CMakeFiles/zigcpp.dir/build.make:76: CMakeFiles/zigcpp.dir/src/zig_llvm.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:91: CMakeFiles/zigcpp.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```
Built with: `cmake .. -DCMAKE_PREFIX_PATH="/usr/lib/llvm-18;/usr/lib" -DCMAKE_BUILD_TYPE=Release`

**Comment by nektro** [comment]
LLVM 18 isnt released yet

**Comment by auxiliaire** [comment]
I can confirm that it works with LLVM 17. This was the source of confusion (from apt.llvm.org):
```
To install the latest stable version:

bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)"
```
Which actually installs LLVM 18.

**Comment by nektro** [comment]
the error you got is additionally odd because cmake should have told you 18 wasnt allowed much sooner than that assert

**Comment by auxiliaire** [comment]
Yeah, I had plenty of such assert. Just quoted the last one. Also had the same error after switching to 14, but that was maybe because I forgot to clean the build dir. I don't know.

**Comment by andrewrk** [comment]
Fixed with the merge of LLVM 19 branch in c6ad4521c77372cb52adfb9a52b0854d830fed9c.

**Comment by Rexicon226** [comment]
This should not be closed yet, the test was not re-enabled.
